metaquot/test/dune

```diff
@@ -0,0 +1,5 @@
+(library
+ (name ppx_metaquot_test)
+ (public_name ppx.metaquot_test)
```

Comment:
You should make it a test stanza instead so this is not a public library therefore it's not installed and it's only built when we run tests.
I.e. you should turn it into:
```
(test
 (name test_ppx_metaquot)
 (package ppx)
 (libraries ppx)
 (preprocess (pps ppx.metaquot ppx.runner)))
```

Reply:
Hmm. If I had multiple test modules in a directory, would I have to create one test clause for each? Is there a way to say "all the ML files in this directory are tests" like we say "all the ML files in this directory are part of a library"?

Reply:
We could have a private library and a test executable depending on it:
```
(libary
 (name ppx_metaquot_test)
 (libraries ppx)
 (preprocess (pps ppx.metaquot)))

(test
 (name test)
 (modules test)
 (package ppx)
 (libraries ppx_metaquot_test))
```

The test executable can be empty as the tests consists of making sure the library builds. This combine the advantages of both approach, i.e. build all modules but only as part of the `@runtest` alias and don't install any of this.

You might want to move the dummy test binary or the lib to a subfolder to avoid having to set the `modules` field for either of them and get rid of the "inaccurate merlin" warning!

How does that sound?

Reply:
It might be worth opening an issue on dune to describe this setup and see if there's a better way to do this or if they'd be willing to support it explicitly.

CC @diml @rgrinberg

Reply:
AFAIU, we just want to test that the code compiles, right? It is not a very common case, in general we also want to run things, so I don't think it is worth a special case in Dune. The split with a library seems fine to me. You can also have just a library and this:

```ocaml
(alias
 (name runtest)
 (deps (alias check)))
```

Reply:
Yeah the deps between `runtest` and check seems better indeed, no need for a dummy test executable then!